### PR TITLE
[W09H01] Some utility tests for `kidFriendlyTrip()` and `pathLenghtSimple()` and for nearly every other function in this homework

### PR DESCRIPTION
**Pull request by Benn1x**
The Test checks at first if `pathLenghtSimple()` works with a simple oneWay and second time with 2 oneWays 
For `kidFriendlyTrip()` i just used the example from artemis.
Ig the `assertArrayEquals()` method i wrote works

### CODE REVIEW

**Review by MaximilianAnzinger — APPROVED** [review]

**Comment by Benn1x** [comment]
I added a few more tests for `furthestAWay()` und `transformToWays()`

**Comment by Benn1x** [comment]
Here we go again

**Comment by Benn1x** [comment]
ig now I´m done with passing all these test i passed all artemis test so ig they are right

**Review by nilsreichardt — CHANGES_REQUESTED** [review]
Hey! 👋  Thanks for your PR!

Could you split all the tests for the tasks into smaller files? For each task a new file, like I did with `ReadWayPoints` and `TransformToWays` (see [here](https://github.com/MaximilianAnzinger/pgdp2223-tests/tree/main/w09h01/test/pgdp/pingutrip)).

The advantage of this is that you can easier run all the tests for one task. Otherwise, you are going to execute all the tests for all tasks which haven't implemented yet.

**Comment by Benn1x** [comment]
@nilsreichardt i hope know its ok i the test still works and i noticed that there are now 2 transformToWaysTests should i merge these files to getter?

**Comment by nilsreichardt** [comment]
> @nilsreichardt i hope know its ok i the test still works and i noticed that there are now 2 transformToWaysTests should i merge these files to getter?

Yes, that would be great - thanks! I'm going later to review a second time 👍

**Comment by Benn1x** [comment]
@nilsreichardt i hope now is everything right

File: w09h01/test/pgdp/pingutrip/kidFriedlyTripTest.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package pgdp.pingutrip;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.fail;
```
Comment (nilsreichardt) [inline w09h01/test/pgdp/pingutrip/kidFriedlyTripTest.java:8]:
```suggestion
import static org.junit.jupiter.api.Assertions.assertArrayEquals;
```

We can use the built-in `assertArrayEquals` method

File: w09h01/test/pgdp/pingutrip/kidFriedlyTripTest.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    private void assertArrayEquals(List<OneWay> kidFriendlyTrip, List<OneWay> result) {
+        if(kidFriendlyTrip.size() != result.size()) fail("Unequal already in size. Expected: "+result.size()+" but were: "+kidFriendlyTrip.size());
+        else{
+            for (int i = 0; i < kidFriendlyTrip.size(); i++) {
+                if(!kidFriendlyTrip.get(i).equals(result.get(i))) fail("Unequal at element: "+i+" your were: "+kidFriendlyTrip.get(i)+" the expectet where: "+result.get(i));
+            }
+        }
+    }
```
Comment (nilsreichardt) [inline w09h01/test/pgdp/pingutrip/kidFriedlyTripTest.java:44]:
```suggestion
```

Code is not used anymore

File: w09h01/test/pgdp/pingutrip/kidFriedlyTripTest.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+public class kidFriedlyTripTest {
+    @Test
+    @DisplayName("Checks if really stops in kidFriendlyTripTest when the first element is longer then the average ")
+    void kidFriendlyTripTest(){
+        List<OneWay> lists = List.of(new OneWay(new WayPoint(0.0, 0.0), new WayPoint(4.0, 0.0)),
+                new OneWay(new WayPoint(0.0, 0.0), new WayPoint(1.0, 0.0)),
+                new OneWay(new WayPoint(1.0, 0.0), new WayPoint(3.0, 0.0)));
+        List<OneWay> result = List.of();
+        assertArrayEquals(PinguTrip.kidFriendlyTrip(lists), result);
+    }
```
Comment (nilsreichardt) [inline w09h01/test/pgdp/pingutrip/kidFriedlyTripTest.java:19]:
```suggestion
    void kidFriendlyTripTest() {
        List<OneWay> lists = List.of(new OneWay(new WayPoint(0.0, 0.0), new WayPoint(4.0, 0.0)),
                new OneWay(new WayPoint(0.0, 0.0), new WayPoint(1.0, 0.0)),
                new OneWay(new WayPoint(1.0, 0.0), new WayPoint(3.0, 0.0)));

        List<OneWay> expected = List.of();
        List<OneWay> actual = PinguTrip.kidFriendlyTrip(lists);

        assertArrayEquals(expected.toArray(), actual.toArray());
    }
```

This is more readable 👍

File: w09h01/test/pgdp/pingutrip/kidFriedlyTripTest.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    @DisplayName("Checks if every element gets returned, checks for the correctness of the math")
+    void kidFriendlyTripTest2(){
+        List<OneWay> lists = List.of(new OneWay(new WayPoint(2.0, 0.0), new WayPoint(4.0, 0.0)),
+                new OneWay(new WayPoint(0.0, 0.0), new WayPoint(2.0, 0.0)),
+                new OneWay(new WayPoint(0.0, 0.0), new WayPoint(2.0, 0.0)));
+        List<OneWay> result = List.of(new OneWay(new WayPoint(2.0, 0.0), new WayPoint(4.0, 0.0)),
+                new OneWay(new WayPoint(0.0, 0.0), new WayPoint(2.0, 0.0)),
+                new OneWay(new WayPoint(0.0, 0.0), new WayPoint(2.0, 0.0)));
+        assertArrayEquals(PinguTrip.kidFriendlyTrip(lists), result);
+    }
```
Comment (nilsreichardt) [inline w09h01/test/pgdp/pingutrip/kidFriedlyTripTest.java:31]:
```suggestion
    void kidFriendlyTripTest2() {
        List<OneWay> lists = List.of(new OneWay(new WayPoint(2.0, 0.0), new WayPoint(4.0, 0.0)),
                new OneWay(new WayPoint(0.0, 0.0), new WayPoint(2.0, 0.0)),
                new OneWay(new WayPoint(0.0, 0.0), new WayPoint(2.0, 0.0)));

        List<OneWay> expected = List.of(new OneWay(new WayPoint(2.0, 0.0), new WayPoint(4.0, 0.0)),
                new OneWay(new WayPoint(0.0, 0.0), new WayPoint(2.0, 0.0)),
                new OneWay(new WayPoint(0.0, 0.0), new WayPoint(2.0, 0.0)));
        List<OneWay> actual = PinguTrip.kidFriendlyTrip(lists);

        assertArrayEquals(expected.toArray(), actual.toArray());
    }
```

This is more readable 👍

**Review by nilsreichardt — CHANGES_REQUESTED** [review]
Thanks for the update! 🙏 I added a few more comments. However, when we have resolved the comments, we can merge this 👍

**Comment by Benn1x** [comment]
@nilsreichardt i have commited the suggestions

**Comment by nilsreichardt** [comment]
> i have commited the suggestions

Have you also pushed them? I'm not seeing them on GitHub o.O

**Comment by Benn1x** [comment]
> > i have commited the suggestions
> 
> Have you also pushed them? I'm not seeing them on GitHub o.O

Nope

**Comment by Benn1x** [comment]
But now i have comitted @nilsreichardt

**Review by nilsreichardt — APPROVED** [review]
LGTM ✅

@Benn1x FYI: There were still bugs in your Code. However, I fixed them for your that we can merge it now 👍

**Comment by Benn1x** [comment]
@nilsreichardt Thx fors debugging

**Pull request merged**